org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtDependenciesAnalysis.scala

```diff
@@ -0,0 +1,16 @@
+package org.scalaide.core.internal.builder.zinc
+
+import sbt.inc.IncOptions
+import sbt.inc.Analysis
+import org.scalaide.core.internal.builder.EclipseBuildManager
+import org.eclipse.core.resources.IFile
+
+trait SbtDependenciesAnalysis {
+  def latestAnalysis(incOptions: => IncOptions): Analysis
+}
+
+trait CachedSbtDependenciesAnalysis extends SbtDependenciesAnalysis {
+  def analysisStore: IFile
+}
+
+trait SbtAwareBuildManager extends EclipseBuildManager with SbtDependenciesAnalysis
```

Comment:
I'd like to suggest a long-overdue refactoring: merge `EclipseBuildManager` with `EclipseSbtBuildManager`. The reason to have this abstraction was that for a long time we had two different implementations: a very fast (and sometimes unstable) builder based on the Scala compiler directly, and the Sbt one. Now we only support the Sbt builder, so there's no point in keeping the abstract interface and the `Sbt` implementation. Now, this code, adds a third abstraction just to add a method `latestAnalysis`.

I think all three should be collapsed, and logic is simplified down the line (in all places where we match for the build manager actually being `EclipseSbtBuildManager`).